### DISCUSSION TITLE
Redirect loop with Codespaces beta

### DISCUSSION THREAD

**Discussion by hongshaoyang** [community/community#26897] [Codespaces]
<p>I’ve just joined the Codespaces beta. On clicking on the link to the workspace, it goes into a redirect loop with these 2 URLs:</p>
<ul>
<li><a href="https://github.com/codespaces/hongshaoyang-ipython-2p3p" rel="nofollow noopener">https://github.com/codespaces/hongshaoyang-ipython-2p3p</a></li>
<li><a href="https://hongshaoyang-ipython-2p3p.github.dev/" rel="nofollow noopener">https://hongshaoyang-ipython-2p3p.github.dev/</a></li>
</ul>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4c68b0185e4def9c1f3dd73523d1818ddf4369fb.png" data-download-href="/uploads/short-url/aTWzZnQTrW2CJw1ekTZtUfNTDCP.png?dl=1" title="Screenshot 2020-09-12 at 9.47.01 PM" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/19281800/181100507-8ef26253-86b0-4533-9783-ca17917d171c.png" alt="Screenshot 2020-09-12 at 9.47.01 PM" data-base62-sha1="aTWzZnQTrW2CJw1ekTZtUfNTDCP" width="690" height="276" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/4/4c68b0185e4def9c1f3dd73523d1818ddf4369fb_2_690x276.png, https://aws1.discourse-cdn.com/github/optimized/2X/4/4c68b0185e4def9c1f3dd73523d1818ddf4369fb_2_1035x414.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/4/4c68b0185e4def9c1f3dd73523d1818ddf4369fb_2_1380x552.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4c68b0185e4def9c1f3dd73523d1818ddf4369fb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-09-12 at 9.47.01 PM</span><span class="informations">2784×1114 730 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>I am on Chrome 85.</p>

**Comment by legomushroom** [comment]
<p>Hey @ <a href="https://github.community/u/hongshaoyang">hongshaoyang</a>, thanks for reporting this!</p>
<p>Looks like the browser cannot set either cookie or localStorage record. Is the loop infinite? If it is, that might suggest issues with localStorage.</p>
<p>Can you see if what’s in cookies/localstorage in DevTools -&gt; Application tab? We expect <code>__Host-vssaas.session</code> cookie and <code>vsc-partner-info</code> localstorage record to be saved.</p>
<p>Also can you try incognito window without browser extensions?</p>

**Comment by hongshaoyang** [comment]
<p>It looks like <code> __Host-vssaas.session</code> is set but not <code>vsc-partner-info</code>. I’ve also tried incognito without extensions and tried to change Chrome’s cooking settings to allow all cookies, but neither of these work.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/5/5b6b9ac413aebd4fae6dcbdf1ffe450e74c42059.png" data-download-href="/uploads/short-url/d2JZtlv5cdJ80bxxMRqcqEOLNXz.png?dl=1" title="Screenshot 2020-09-13 at 2.59.03 PM" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/19281800/181100509-d2ed7deb-3a93-47a7-a427-80198b6d56c6.png" alt="Screenshot 2020-09-13 at 2.59.03 PM" data-base62-sha1="d2JZtlv5cdJ80bxxMRqcqEOLNXz" width="690" height="64" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/5/5b6b9ac413aebd4fae6dcbdf1ffe450e74c42059_2_690x64.png, https://aws1.discourse-cdn.com/github/optimized/2X/5/5b6b9ac413aebd4fae6dcbdf1ffe450e74c42059_2_1035x96.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/5/5b6b9ac413aebd4fae6dcbdf1ffe450e74c42059_2_1380x128.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/5/5b6b9ac413aebd4fae6dcbdf1ffe450e74c42059_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot 2020-09-13 at 2.59.03 PM</span><span class="informations">2550×240 45.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by legomushroom** [comment]
<p><a class="mention" href="/hongshaoyang">@hongshaoyang</a> thanks for the reply, that scopes down the issue a bit. Could you do this:</p>
<ol>
<li>Open the <code>https://github.com/codespaces</code> in a chromium-based browser.</li>
<li>When it connections, pop up the DevTools and switch to the <code>Network</code> tab, then check the “Preserve log” checkbox above the network requests table.</li>
<li>Click on a Codespace link so it will try to connect to the Codespace.</li>
<li>Click on the <code>Export HAR</code> arrow and send the archive to <a href="mailto:legomushroom@github.com">legomushroom@github.com</a> <strong>Please don’t attach it to this thread as it might contain sensitive information.</strong>
</li>
</ol>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/cb0d2a604c7643608ff156719458777c1b2892a7.jpeg" data-download-href="/uploads/short-url/sYh8hgjaSZDphbftJKsWXVQkWaj.jpeg?dl=1" title="export" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/1478800/181100511-2c7f86f4-8ced-4501-b1f6-73f5cbf83cb4.jpeg" alt="export" data-base62-sha1="sYh8hgjaSZDphbftJKsWXVQkWaj" width="690" height="435" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/c/cb0d2a604c7643608ff156719458777c1b2892a7_2_690x435.jpeg, https://aws1.discourse-cdn.com/github/optimized/2X/c/cb0d2a604c7643608ff156719458777c1b2892a7_2_1035x652.jpeg 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/c/cb0d2a604c7643608ff156719458777c1b2892a7_2_1380x870.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/cb0d2a604c7643608ff156719458777c1b2892a7_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">export</span><span class="informations">1898×1199 322 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>That would give me the ability to look at what happens at the authorization step in more detail.</p>

**Comment by hongshaoyang** [comment]
<p>Thanks <a class="mention" href="/legomushroom">@legomushroom</a>, have sent you an email with the HAR file.</p>

**Comment by legomushroom** [comment]
<p><a class="mention" href="/hongshaoyang">@hongshaoyang</a>, thanks for the info! The data you’ve sent to me is <code>perfectly fine</code> so it scopes down the issue a bit more.</p>
<p>Few questions:</p>
<ul>
<li>Does it fail in all browsers you have or just one?</li>
<li>Are there any errors in the DevTools console? (please send them to me if you see anything there)</li>
<li>Do you have any records in the LocalStorage at all or there are some and only the <code>vsc-partner-info</code> record is missing?</li>
</ul>
<p>I’ll dive into logging to see if I can catch anything interesting. If that does not help, do you think we can have a screen share session to debug this together?</p>

**Comment by hongshaoyang** [comment]
<ul>
<li>I tried with Firefox Developer Edition 81 and it worked fine, so it might be a Chrome issue.</li>
<li>The only error i see is <code>stats.ts:41 POST https://api.github.com/_private/browser/stats net::ERR_BLOCKED_BY_CLIENT</code> which I believe is just sending browser stats and is blocked by uBlock origin.</li>
<li>LocalStorage has one record: <code>vscs-oauth-flow-attmept-count: 1</code> [<em>sic</em>]</li>
</ul>
<p>Yes, I am fine with a screenshare session to debug this.</p>

**Comment by hongshaoyang** [comment]
<p>Hold on, there’s actually another error in the console:</p>
<pre><code class="lang-auto">VM36 workbench.web.api.nls.js:4 Uncaught ReferenceError: define is not defined
    at VM36 workbench.web.api.nls.js:4
(anonymous) @ VM36 workbench.web.api.nls.js:4
</code></pre>
<p>Could this be relevant?</p>

**Comment by legomushroom** [comment]
<p>Thanks for trying out all of these :+1:</p>
<p>Both errors don’t look related, the fact that you have just a single LocalStorage record is very interesting. I suspect that is an issue with LocalStorage or WebCrypto APIs in Chrome but would be great to put my hands on the live browser in a screen share session since I cannot repro it on my end.</p>
<p>I see your profile says you are in Singapore and it is past midnight there, please ping me by email when you back online so we can arrange the debugging session. Thanks a lot!</p>

**Comment by legomushroom** [comment, marked as answer]
<p>Just finished the screenshare debugging session with <a href="https://github.community/u/hongshaoyang">hongshaoyang</a>.<br>
Looks like the <a href="https://github.com/iamadamdev/bypass-paywalls-chrome" rel="noopener nofollow ugc">https://github.com/iamadamdev/bypass-paywalls-chrome</a> extension was blocking/removing <code>localStorage</code> records.</p>
